ci/scripts/cpp_test.sh

```diff
@@ -80,6 +80,12 @@ pushd ${build_dir}
 if [ -z "${PYTHON}" ] && ! which python > /dev/null 2>&1; then
   export PYTHON="${PYTHON:-python3}"
 fi
+if [[ "${ARROW_ENABLE_THREADING}" == "OFF" ]]; then
+# if threading is disabled, some tests take longer to run
+# but we can get away with running more tests in parallel because we know they wont make loads of threads
+  ARROW_CTEST_TIMEOUT=${ARROW_CTEST_TIMEOUT:-900}
```

Comment:
It seems that max test time is "239.99 sec" in our CI:

https://github.com/ursacomputing/crossbow/actions/runs/5018372445/jobs/8997694994#step:6:3234

> ```text
> 89/103 Test  #21: arrow-compute-scalar-if-else-test .........   Passed  239.99 sec
> ```

Why is this needed?